opentracing-api/src/main/java/io/opentracing/ActiveSpanSource.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing;
+
+/**
+ * {@link ActiveSpanSource} allows an existing (possibly thread-local-aware) execution context provider to act as a
+ * source for an actively-scheduled OpenTracing Span.
+ *
+ * @see ActiveSpan
+ */
+public interface ActiveSpanSource {
+
+    /**
+     * Return the {@link ActiveSpan active span}. This does not affect the reference count for the {@link ActiveSpan}.
+     *
+     * <p>
+     * If there is an {@link ActiveSpan active span}, it becomes an implicit parent
+     * at {@link Tracer.SpanBuilder#startActive()} time, not at {@link Tracer#buildSpan(String)} time.
+     *
+     * @return the {@link ActiveSpan active span}, or null if none could be found.
+     */
+    ActiveSpan activeSpan();
```

Comment:
I'm not sure if this has already been discussed, but from a user's standpoint I'd prefer returning a `NoopSpan` over returning `null` so that they don't have to do null checks just to add a simple tag.

If the tag is expensive to compute, it would however be beneficial if the user could differ between proper and noop spans. But rather than checking for `null`s or a global `NOOP` constant I propose to introduce a `isSampled()` method on `BaseSpan`. From the user's perspective, it does not really matter wether there is no active span or the active span is not sampled. They only want to avoid computing a expensive tag if not necessary.

I think this is something we should decide on before merging, or at least before releasing a new version as this would be a somewhat breaking change (although remaining binary compatibility) to introduce later (check for `null` vs check for `isSampled()`).